vllm/worker/worker.py

```diff
@@ -36,6 +36,7 @@ def __init__(
         rank: int,
         distributed_init_method: str,
         lora_config: Optional[LoRAConfig] = None,
+        cache_config: Optional[CacheConfig] = None,
```

Comment:
This change is weird. Originally we set the `cache_config` in `self.init_cache_engine()` (as in line 60 below). This change introduces two `cache_config` objects, which is super confusing.

The reason we delay the initialization of the `cache_config` is that `cache_config` includes the number of KV blocks, which can only be known after memory profiling.

To make things more clear, I think we can just feed in `kv_cache_dtype` here.

Reply:
Your suggestion is good for me. `cache_config: Optional[CacheConfig] = None`  ->  `kv_cache_dtype: Optional[str] = "auto"`